src/vm/typestring.cpp

```diff
@@ -979,17 +979,22 @@ void TypeString::AppendType(TypeNameBuilder& tnb, TypeHandle ty, Instantiation t
         // Get the TypeDef token and attributes
         IMDInternalImport *pImport = ty.GetMethodTable()->GetMDImport();
         mdTypeDef td = ty.GetCl();
-        _ASSERTE(!IsNilToken(td));
-
-#ifdef _DEBUG
-        if (format & FormatDebug)
-        {
-            WCHAR wzAddress[128];
-            _snwprintf_s(wzAddress, 128, _TRUNCATE, W("(%p)"), dac_cast<TADDR>(ty.AsPtr()));
-            tnb.AddName(wzAddress);
+        if (IsNilToken(td)) {
+          // This type does not exist in metadata. Simply append "dynamicClass".
+          tnb.AddName(W("(dynamicClass)"));
         }
```

Comment:
I see. I will leave it up to you which one you like better. Either one will be fine with me.